src/Service/Nph/NphOrderService.php

```diff
@@ -597,6 +597,32 @@ public function saveSamplesModification(array $formData, string $type, NphOrder
         return $status;
     }
 
+    public function updateSampleModificationBulk(array $formData, string $type): bool
+    {
+        $status = true;
+        foreach ($formData as $sampleId => $checked) {
+            $sample = $this->em->getRepository(NphSample::class)->findOneBy(['sampleId' => $sampleId]);
+            if ($checked) {
+                $sampleObject = $this->getCancelRestoreRdrObject($type, $formData['reason']);
+                if ($sample->getRdrId()) {
```

Comment:
RDR id is only available for samples that are sent to RDR. You might need to handle the samples that do not have the RDR id set.